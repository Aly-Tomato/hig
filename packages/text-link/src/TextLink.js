import { TextLink } from "hig-react";
import "./text-link.scss";

TextLink.displayName = "TextLink";

export default TextLink;
