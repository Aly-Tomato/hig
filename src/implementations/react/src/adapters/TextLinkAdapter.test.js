
import { mount } from 'enzyme';
import * as HIG from 'hig-vanilla';
import React from 'react';

import TextLink from './TextLinkAdapter';

describe('<TextLinkAdapter>', () => {
  /**
     * Creates a hig-vanilla text link and returns the instance and it's container
     *
     * @param {object} defaults hig-vanilla defaults for the text link
     */
  function createHigTextLink(defaults = {}) {
    const higContainer = document.createElement('div');

    // use spread here to clone defaults since HIG.Button mutates this object
    const higTextLink = new HIG.TextLink({ ...defaults });

    higTextLink.mount(higContainer);

    return { higTextLink, higContainer };
  }

  it('renders a primary text link', () => {
    const defaults = { text: 'this is a primary text link', href: 'http://example.com' };

    const { higTextLink, higContainer } = createHigTextLink(defaults);
    const container = document.createElement('div');

    const wrapper = mount(<TextLink {...defaults} />, { attachTo: container });

    expect(container.firstElementChild.outerHTML).toMatchSnapshot();

    expect(container.firstElementChild.outerHTML).toEqual(
      higContainer.firstElementChild.outerHTML
    );
  });

  it('renders a secondary text link', () => {
    const defaults = { text: 'this is a secondary text link', href: 'http://example.com', type: 'secondary' };

    const { higTextLink, higContainer } = createHigTextLink(defaults);
    const container = document.createElement('div');

    const wrapper = mount(<TextLink {...defaults} />, { attachTo: container });

    expect(container.firstElementChild.outerHTML).toMatchSnapshot();

    expect(container.firstElementChild.outerHTML).toEqual(
      higContainer.firstElementChild.outerHTML
    );
  });

});
