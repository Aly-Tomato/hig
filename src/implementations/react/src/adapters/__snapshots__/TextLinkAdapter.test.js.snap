// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextLinkAdapter> renders a primary text link 1`] = `"<a class=\\"hig__text-link hig__text-link--primary\\" href=\\"http://example.com\\" alt=\\"this is a primary text link\\">this is a primary text link</a>"`;

exports[`<TextLinkAdapter> renders a secondary text link 1`] = `"<a class=\\"hig__text-link hig__text-link--secondary\\" href=\\"http://example.com\\" alt=\\"this is a secondary text link\\">this is a secondary text link</a>"`;
